### PR TITLE
add & remove user permission

### PR DESCRIPTION
**Pull request by ay0o5h**
add & remove  user permission in api gateway

### CODE REVIEW

**Review by HaidyAbuGom3a — APPROVED** [review]

File: api_gateway/src/main/kotlin/org/thechance/api_gateway/endpoints/PermissionRoute.kt [hunk 1]
```diff
@@ -14,8 +15,32 @@ import org.thechance.api_gateway.util.Role
...
 
+    route("/dashboard"){
+
+            authenticateWithRole(Role.DASHBOARD_ADMIN) {
+                get("/restaurant-request") {
+                    val language = extractLocalizationHeader()
+                    val result = restaurantService.getAllRequestPermission(language)
+                    respondWithResult(HttpStatusCode.OK, result)
+                }
+                post("user/{userId}/permission"){
```
Comment (NadaFeteiha) [inline api_gateway/src/main/kotlin/org/thechance/api_gateway/endpoints/PermissionRoute.kt:29]: please add it to Route.userRoutes()

File: api_gateway/src/main/kotlin/org/thechance/api_gateway/endpoints/PermissionRoute.kt [hunk 2]
```diff
@@ -14,8 +15,32 @@ import org.thechance.api_gateway.util.Role
...
+                    val userId = call.parameters["userId"]?.trim().toString()
+                    val permission = call.receiveParameters()["permission"]?.trim()?.toInt() ?: 0
+                    val result = identityService.updateUserPermission(userId,permission)
+                    respondWithResult(HttpStatusCode.Created, result)
+                }
+                put("user/{userId}/permission"){
```
Comment (NadaFeteiha) [inline api_gateway/src/main/kotlin/org/thechance/api_gateway/endpoints/PermissionRoute.kt:35]: please add it to Route.userRoutes()

**Review by NadaFeteiha — CHANGES_REQUESTED** [review]

File: api_gateway/src/main/kotlin/org/thechance/api_gateway/endpoints/UserRoute.kt [hunk 3]
```diff
@@ -38,7 +40,21 @@ fun Route.userRoutes() {
                 val result = identityService.deleteUser(userId = id, language)
                 respondWithResult(HttpStatusCode.OK, result)
             }
+            post("/{userId}/permission"){
+                val userId = call.parameters["userId"]?.trim().toString()
+                val permission = call.receiveParameters()["permission"]?.trim()?.toInt() ?: 0
```
Comment (NadaFeteiha) [inline api_gateway/src/main/kotlin/org/thechance/api_gateway/endpoints/UserRoute.kt:45]: "I think this is the same as the filtering we discussed in the meeting, where we talked about being able to filter by multiple permissions, not just one.

File: api_gateway/src/main/kotlin/org/thechance/api_gateway/endpoints/PermissionRoute.kt [hunk 4]
```diff
@@ -14,8 +15,20 @@ import org.thechance.api_gateway.util.Role
 
 fun Route.permissionRoutes() {
 
+    val identityService: IdentityService by inject()
     val restaurantService: RestaurantService by inject()
 
+    route("/dashboard"){
+
+            authenticateWithRole(Role.DASHBOARD_ADMIN) {
+                get("/restaurant-request") {
```
Comment (NadaFeteiha) [inline api_gateway/src/main/kotlin/org/thechance/api_gateway/endpoints/PermissionRoute.kt:24]: I believe this route was added by mistake and is not being used. You can find it in the develop and main branches, under the route ("/permission")

**Review by NadaFeteiha — CHANGES_REQUESTED** [review]

File: api_gateway/src/main/kotlin/org/thechance/api_gateway/endpoints/UserRoute.kt (hunk 3 above)
Comment (ay0o5h) [reply api_gateway/src/main/kotlin/org/thechance/api_gateway/endpoints/UserRoute.kt:45]: this is not a search  just update permission @NadaFeteiha

File: api_gateway/src/main/kotlin/org/thechance/api_gateway/endpoints/PermissionRoute.kt (hunk 4 above)
Comment (ay0o5h) [reply api_gateway/src/main/kotlin/org/thechance/api_gateway/endpoints/PermissionRoute.kt:24]: i didn't add it was already there

**Review by mustafa-sab3alarab — APPROVED** [review]

**Review by NadaFeteiha — APPROVED** [review]

**Pull request merged**